datashader/core.py

```diff
@@ -5,7 +5,10 @@
 from odo import discover
 from xarray import DataArray
 
-from .utils import Dispatcher, ngjit
+from .utils import Dispatcher, ngjit, calc_res, calc_bbox
+from .resampling import (resample_2d, US_NEAREST, US_LINEAR, DS_FIRST, DS_LAST,
+                         DS_MEAN, DS_MODE, DS_VAR, DS_STD)
+
```

Comment:
I'd rather not import all those here, with names like that; it's not pythonic.  Instead, can you have `resampling` provide the `upsample_methods` and `downsample_methods` dictionaries?  In fact, we should just go ahead and make these classes Parameterized, as at the end of bundling.py, so that these values can be expressed directly; no need for the special logic that is in this file to handle how to drive some other file.